applications/DamApplication/custom_constitutive/thermal_linear_elastic_3D_law_nodal.cpp

```diff
@@ -200,7 +200,7 @@ double&  ThermalLinearElastic3DLawNodal::CalculateNodalReferenceTemperature (con
 
     for ( unsigned int j = 0; j < number_of_nodes; j++ )
     {
-      rNodalReferenceTemperature += ShapeFunctionsValues[j] * DomainGeometry[j].GetSolutionStepValue(NODAL_REFERENCE_TEMPERATURE);
+      rNodalReferenceTemperature += ShapeFunctionsValues[j] * DomainGeometry[j].GetSolutionStepValue(REFERENCE_TEMPERATURE);
```

Comment:
we should check somewhere 

      node.HasSoltuionStepValue(REFERENCE_TEMPERATURE)

and otherwise avoid this loop

Reply:
wouldn't this one be redundant if the behaviour is in the classic CL?

Reply:
This Cl is not from the Structural Mech App. In the case of the LinearElastic from the SMA we are doing the checks

Reply:
Done!